File: PHP/02_calculadora.php
<?php
/**
 * 02 - Calculadora
 * Conceitos: if/elseif, switch, operadores, validação
 */

function somar($a, $b) { return $a + $b; }
function subtrair($a, $b) { return $a - $b; }
function multiplicar($a, $b) { return $a * $b; }

function dividir($a, $b) {
    if ($b == 0) {
        return "Erro: divisão por zero!";
    }
    return $a / $b;
}

echo "=== CALCULADORA ===\n";
echo "1 - Somar\n";
echo "2 - Subtrair\n";
echo "3 - Multiplicar\n";
echo "4 - Dividir\n";
echo "Escolha uma opção (1-4): ";
$opcao = trim(fgets(STDIN));

echo "Digite o primeiro número: ";
$num1 = (float) trim(fgets(STDIN));

echo "Digite o segundo número: ";
$num2 = (float) trim(fgets(STDIN));

switch ($opcao) {
    case "1":
        echo "Resultado: " . somar($num1, $num2) . "\n";
        break;
    case "2":
        echo "Resultado: " . subtrair($num1, $num2) . "\n";
        break;
    case "3":
        echo "Resultado: " . multiplicar($num1, $num2) . "\n";
        break;
    case "4":
        echo "Resultado: " . dividir($num1, $num2) . "\n";
        break;
    default:
        echo "Opção inválida!\n";
}

File: Python/02_calculadora.py
"""
02 - Calculadora
Conceitos: funções, operadores, if/elif/else, try/except
"""

def somar(a, b):
    return a + b

def subtrair(a, b):
    return a - b

def multiplicar(a, b):
    return a * b

def dividir(a, b):
    if b == 0:
        return "Erro: divisão por zero!"
    return a / b

def calculadora():
    print("\n=== CALCULADORA ===")
    print("1 - Somar")
    print("2 - Subtrair")
    print("3 - Multiplicar")
    print("4 - Dividir")

    opcao = input("Escolha uma opção (1-4): ")

    try:
        num1 = float(input("Digite o primeiro número: "))
        num2 = float(input("Digite o segundo número: "))
    except ValueError:
        print("Entrada inválida! Digite apenas números.")
        return

    if opcao == "1":
        print(f"Resultado: {somar(num1, num2)}")
    elif opcao == "2":
        print(f"Resultado: {subtrair(num1, num2)}")
    elif opcao == "3":
        print(f"Resultado: {multiplicar(num1, num2)}")
    elif opcao == "4":
        print(f"Resultado: {dividir(num1, num2)}")
    else:
        print("Opção inválida!")

if __name__ == "__main__":
    calculadora()